extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxBlockingOutput.java

```diff
@@ -89,6 +92,44 @@ public void write(ByteBuf data, boolean last) throws IOException {
         }
     }
 
+    @Override
+    public CompletionStage<Void> writeNonBlocking(ByteBuf data, boolean last) {
+        CompletableFuture<Void> ret = new CompletableFuture<>();
+        if (last && data == null) {
+            request.response().end(handler(ret));
+            return ret;
+        }
+        Buffer buffer = createBuffer(data);
+        if (last) {
+            request.response().end(buffer, handler(ret));
+        } else {
+            request.response().write(buffer, handler(ret));
+        }
+        return ret.handle((v, t) -> {
+            if (t != null) {
+                if (data != null && data.refCnt() > 0) {
+                    data.release();
+                }
+                rethrow(new IOException("Failed to write", t));
```

Comment:
This `rethrow` method seems useless if you use `UncheckedIOException` instead of the checked version

Reply:
Well, only if the caller knows to unwrap them. I prefer to not rely on that.